pkg/kapp/cmd/app/deploy.go

```diff
@@ -400,6 +404,11 @@ func (o *DeployOptions) calculateAndPresentChanges(existingResources,
 		changeFactory := ctldiff.NewChangeFactory(conf.RebaseMods(), conf.DiffAgainstLastAppliedFieldExclusionMods())
 		changeSetFactory := ctldiff.NewChangeSetFactory(o.DiffFlags.ChangeSetOpts, changeFactory)
 
+		err := CheckForPriodicAnnotation(existingResources, newResources)
```

Comment:
Couple of points here:
- The function name suggests that we are checking for `PriodicAnnotation`, but we are also making some updates to the resources based on the presence of that annotation, so the name should signify that update as well.
- It seems a bit weird to have this logic present here. It seems more like a _preparation_ we would want to do before calculating the changes, so maybe it should be present in the diff package.